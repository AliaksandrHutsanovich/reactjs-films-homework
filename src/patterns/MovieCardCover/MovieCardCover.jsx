import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Layout } from 'antd';
import MovieInfoCard from '../MovieInfoCard';

const MovieCardCover = ({
  title,
  genres,
  rating,
  info,
  onClose,
  imgUrl,
  layoutClassName,
  onDialogOpen,
}) => (
  <Layout
    style={{
      background: `url(${imgUrl}) no-repeat`,
    }}
    className={layoutClassName}
  >
    <MovieInfoCard
      title={title}
      genres={genres}
      rating={rating}
      info={info}
      onClose={onClose}
      onDialogOpen={onDialogOpen}
    />
  </Layout>
);

MovieCardCover.propTypes = {
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  rating: PropTypes.number.isRequired,
  info: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  imgUrl: PropTypes.string.isRequired,
  layoutClassName: PropTypes.string,
  onDialogOpen: PropTypes.func.isRequired,
};

MovieCardCover.defaultProps = {
  layoutClassName: '',
};

export default memo(MovieCardCover);
