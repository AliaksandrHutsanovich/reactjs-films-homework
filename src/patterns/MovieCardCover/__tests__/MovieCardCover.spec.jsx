import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieCardCover from '../MovieCardCover';

describe('<MovieCardCover />', () => {
  const renderer = new ShallowRenderer();
  it('should render properly', () => {
    renderer.render(
      <MovieCardCover
        title="Assasiens criesd"
        genres={['action', 'genre']}
        rating={5.5}
        info="Some info"
        onClose={() => {}}
        imgUrl="//url"
        layoutClassName="class"
        onDialogOpen={() => {}}
      />,
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
