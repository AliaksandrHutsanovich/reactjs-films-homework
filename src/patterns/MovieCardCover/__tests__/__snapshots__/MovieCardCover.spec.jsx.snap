// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardCover /> should render properly 1`] = `
<Layout
  className="class"
  style={
    Object {
      "background": "url(//url) no-repeat",
    }
  }
>
  <UNDEFINED
    genres={
      Array [
        "action",
        "genre",
      ]
    }
    info="Some info"
    onClose={[Function]}
    onDialogOpen={[Function]}
    rating={5.5}
    title="Assasiens criesd"
  />
</Layout>
`;
